pkg/services/accesscontrol/acimpl/service.go

```diff
@@ -373,8 +388,15 @@ func (s *Service) getCachedTeamsPermissions(ctx context.Context, user identity.R
 }
 
 func (s *Service) ClearUserPermissionCache(user identity.Requester) {
-	s.cache.Delete(accesscontrol.GetPermissionCacheKey(user))
+	s.cache.Delete(accesscontrol.GetUserPermissionCacheKey(user))
 	s.cache.Delete(accesscontrol.GetUserDirectPermissionCacheKey(user))
+
+	// Clear user's teams cache
+	teams := user.GetTeams()
+	orgID := user.GetOrgID()
+	for _, teamID := range teams {
+		s.cache.Delete(accesscontrol.GetTeamPermissionCacheKey(teamID, orgID))
+	}
```

Comment:
From what I can see, we only call `ClearUserPermissionCache` when:
* a new dash/folder/data source/service account/team is created;
* user's basic org role is updated.

In none of these scenarios the perms granted to the existing teams change. So I think it's fine to not clear team perm cache here. But I do think we should be better at clearing caches in general (eg, if a team is granted a new permission).

Reply:
Yeah, I also thought about this. It basically has nothing to do with teams permissions. I'll remove this.